### PR TITLE
 Merging aws-android-sdk-amazon-freerto

### PR DESCRIPTION
**Pull request by bxpan**
*Issue #, if available:*

*Description of changes:*

Merging aws-android-sdk-amazon-freerto  from commit "Add BLE scan fail errorcode callback" https://code.amazon.com/packages/Amazon-freertos-ble-android-sdk/commits/63b25022b56bef1022b6bc92b5eaa6a3bcf5bfe0#

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-android-sdk-amazon-freertos/build.gradle [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+        release {
+            minifyEnabled false
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+
+}
+
+dependencies {
+    implementation 'com.google.code.gson:gson:2.8.5'
```
Comment (minbi) [inline aws-android-sdk-amazon-freertos/build.gradle]: Can this be removed? I don't see usage.

File: aws-android-sdk-amazon-freertos/build.gradle [hunk 2]
```diff
@@ -0,0 +1,37 @@
+apply plugin: 'com.android.library'
+
+android {
+    compileSdkVersion 26
+
+    defaultConfig {
+        minSdkVersion 26
```
Comment (minbi) [inline aws-android-sdk-amazon-freertos/build.gradle]: Can this be lowered to be the actual minimum support?

File: aws-android-sdk-amazon-freertos/build.gradle [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+        }
+    }
+
+}
+
+dependencies {
+    implementation 'com.google.code.gson:gson:2.8.5'
+    implementation 'co.nstant.in:cbor:0.8'
+    implementation ('com.amazonaws:aws-android-sdk-iot:2.12.+')
```
Comment (minbi) [inline aws-android-sdk-amazon-freertos/build.gradle]: implementation project(":aws-android-sdk-iot") {
    exclude group: "com.google.android", module: "android"
}

File: aws-android-sdk-amazon-freertos/build.gradle [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation 'com.android.support:appcompat-v7:26.1.0'
+    testImplementation 'junit:junit:4.12'
+    androidTestImplementation 'com.android.support.test:runner:1.0.2'
+    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
+    compileOnly 'org.projectlombok:lombok:1.18.4'
```
Comment (minbi) [inline aws-android-sdk-amazon-freertos/build.gradle]: Can this be removed? I see very little usage.

File: aws-android-sdk-amazon-freertos/src/main/java/com/amazon/aws/amazonfreertossdk/AmazonFreeRTOSConstants.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
+package com.amazon.aws.amazonfreertossdk;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This class defines some constants used in the SDK.
+ */
+public class AmazonFreeRTOSConstants {
+    public static final String AMAZONFREERTOS_SDK_VERSION = "0.9.3";
```
Comment (minbi) [inline aws-android-sdk-amazon-freertos/src/main/java/com/amazon/aws/amazonfreertossdk/AmazonFreeRTOSConstants.java]: We should change this to be the SDK's version. We have seen the maintenance of this version be difficult and sometimes forgotten.

File: aws-android-sdk-amazon-freertos/src/main/res/values/strings.xml [hunk 6]
```diff
@@ -0,0 +1,3 @@
+<resources>
+    <string name="app_name">AmazonFreeRTOSsdk</string>
```
Comment (minbi) [inline aws-android-sdk-amazon-freertos/src/main/res/values/strings.xml]: This should be removed.

File: aws-android-sdk-amazon-freertos/src/androidTest/java/com/amazon/aws/amazonfreertossdk/ExampleInstrumentedTest.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
+package com.amazon.aws.amazonfreertossdk;
```
Comment (minbi) [inline aws-android-sdk-amazon-freertos/src/androidTest/java/com/amazon/aws/amazonfreertossdk/ExampleInstrumentedTest.java]: The SDKs namespace is `com.amazonaws`

File: aws-android-sdk-amazon-freertos/src/main/java/com/amazon/aws/amazonfreertossdk/AmazonFreeRTOSConstants.java [hunk 8]
```diff
@@ -0,0 +1,116 @@
+package com.amazon.aws.amazonfreertossdk;
```
Comment (minbi) [inline aws-android-sdk-amazon-freertos/src/main/java/com/amazon/aws/amazonfreertossdk/AmazonFreeRTOSConstants.java]: All files are missing license

File: aws-android-sdk-amazon-freertos/build.gradle (hunk 1 above)
Comment (bxpan) [reply aws-android-sdk-amazon-freertos/build.gradle]: Removed

File: aws-android-sdk-amazon-freertos/build.gradle [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+dependencies {
+    implementation 'com.google.code.gson:gson:2.8.5'
+    implementation 'co.nstant.in:cbor:0.8'
+    implementation ('com.amazonaws:aws-android-sdk-iot:2.12.+')
+
+    implementation fileTree(dir: 'libs', include: ['*.jar'])
+    implementation 'com.android.support:appcompat-v7:26.1.0'
+    testImplementation 'junit:junit:4.12'
+    androidTestImplementation 'com.android.support.test:runner:1.0.2'
+    androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
```
Comment (desokroshan) [inline aws-android-sdk-amazon-freertos/build.gradle]: This can be removed since we do not have any espresso tests

**Comment by minbi** [comment]
Closing. No longer merging.

**Pull request closed without merging**